### ISSUE TITLE
Microsoft Visual Studio Failed to launch debug adapter 'Microsoft.VisualStudio.Shared.VSCodeDebugProtocol.ProtocolException

### ISSUE THREAD

**Issue by dazinator** [dotnet/aspnetcore#28113]
### Describe the bug
I am trying to upgrade my .net core 3.1 based solution to .net 5.0.
I am seeing this error with:

- Microsoft Visual Studio Professional 2019 Version 16.8.2
- Microsoft  Visual Studio 2019 Community version 16.9 Preview 1

Attempting to launch my asp.net core project with visual studio debugger fails:

![image](https://user-images.githubusercontent.com/3176632/100107378-7a178800-2e61-11eb-9dc0-49a5d675cd4e.png)

Looking in output window I got a path to the verbose output log, and here is the relevent info:

```json
{"tag":"runtime.launch","timestamp":1606227038803,"message":"Launch returned error","metadata":{"error":{"message":"Unable to launch browser: \"Failed to launch browser!\n\n\nTROUBLESHOOTING: https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md\n\"","stack":"Error: Unable to launch browser: \"Failed to launch browser!\n\n\nTROUBLESHOOTING: https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md\n\"\n at k.prepareLaunch (c:\\program files (x86)\\microsoft visual studio\\2019\\preview\\common7\\ide\\commonextensions\\microsoft\\jsdiagnostics\\debugger\\debugAdapter\\out\\src\\targets\\browser\\browserLauncher.js:102:19)\n at processTicksAndRejections (internal/process/task_queues.js:97:5)"},"wasCancelled":false,"name":"k"},"level":2} {"tag":"dap.send","timestamp":1606227041361,"metadata":{"connectionId":0,"message":{"seq":16,"type":"response","request_seq":2,"command":"launch","success":false,"body":{"error":{"id":9240,"format":"Unable to launch browser: \"Failed to launch browser!\n\n\nTROUBLESHOOTING: https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md\n\"","showUser":false}}}},"level":0} {"tag":"dap.send","timestamp":1606227043301,"metadata":{"connectionId":0,"message":{"seq":17,"type":"event","event":"output","body":{"category":"telemetry","output":"js-debug/dap/operation","data":{"errors":[],"initialize":{"operation":"initialize","totalTime":4.5,"max":4.5,"avg":4.5,"stddev":null,"count":1,"failed":0},"!initialize.errors":[],"initialize.errors":[],"setBreakpoints":{"operation":"setBreakpoints","totalTime":11.3,"max":2,"avg":1.3,"stddev":0.5,"count":9,"failed":0},"!setBreakpoints.errors":[],"setBreakpoints.errors":[],"setExceptionBreakpoints":{"operation":"setExceptionBreakpoints","totalTime":0.5,"max":0.5,"avg":0.5,"stddev":null,"count":1,"failed":0},"!setExceptionBreakpoints.errors":[],"setExceptionBreakpoints.errors":[],"configurationDone":{"operation":"configurationDone","totalTime":0.5,"max":0.5,"avg":0.5,"stddev":null,"count":1,"failed":0},"!configurationDone.errors":[],"configurationDone.errors":[],"launch":{"operation":"launch","totalTime":2932.4,"max":2932.4,"avg":2932.4,"stddev":null,"count":1,"failed":1},"!launch.errors":[{"error":{"message":"Unable to launch browser: \"Failed to launch browser!\n\n\nTROUBLESHOOTING: https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md\n\"","name":"Error","stack":"Error: Unable to launch browser: \"Failed to launch browser!\n\n\nTROUBLESHOOTING: httptroubleshooting.md\n\"\n at k.prepareLaunch (c:browserLauncher.js:102:19)\n at processTicksAndRejections (internaltask_queues.js:97:5)"}}],"launch.errors":[{"error":{"message":"Unable to launch browser: \"Failed to launch browser!\n\n\nTROUBLESHOOTING: https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md\n\"","name":"Error","stack":"Error: Unable to launch browser: \"Failed to launch browser!\n\n\nTROUBLESHOOTING: httptroubleshooting.md\n\"\n at k.prepareLaunch (c:browserLauncher.js:102:19)\n at processTicksAndRejections (internaltask_queues.js:97:5)"}}]}}}},"level":0} {"tag":"dap.send","timestamp":1606227043419,"metadata":{"connectionId":0,"message":{"seq":18,"type":"event","event":"output","body":{"category":"telemetry","output":"js-debug/error","data":{"!error":{"error":{"message":"read ECONNRESET","name":"Error","stack":"Error: read ECONNRESET\n at TCP.onStreamRead (internal/stream_base_commons.js:205:27)"}},"error":{"error":{"message":"read ECONNRESET","name":"Error","stack":"Error: read ECONNRESET\n at TCP.onStreamRead (internal/stream_base_commons.js:205:27)"}},"exceptionType":"uncaughtException"}}}},"level":0} {"tag":"dap.send","timestamp":1606227043419,"message":"Message not sent. Connection was closed.","level":2} {"tag":"runtime.exception","timestamp":1606227043419,"message":"Unhandled error in debug adapter","metadata":{"message":"read ECONNRESET","stack":"Error: read ECONNRESET\n at TCP.onStreamRead (internal/stream_base_commons.js:205:27)"},"level":3}


```

Here is the relevent section from my `launchProfiles.json`:

```json
 "MyHost": {
      "commandName": "Project",
      "launchBrowser": true,
      "launchUrl": "https://localhost:62808/",
      "inspectUri": "{wsProtocol}://{url.hostname}:{url.port}/_framework/debug/ws-proxy?browser={browserInspectUri}",
      "applicationUrl": "http://localhost:62807/;https://localhost:62808/;",
      "environmentVariables": {
        "ASPNETCORE_ENVIRONMENT": "Development"
      }
    },

```

Note: if I change `launchBrowser` to `false` I can successfully start the project, and then open the browser and navigate to the url manually.

Other Note:

Edge is my default browser. When this error occurs, it seems a new empty Edge window is created:

![image](https://user-images.githubusercontent.com/3176632/100107760-ec886800-2e61-11eb-8426-5cab086bd432.png)

### To Reproduce
As per above that's the best I can do at present.

### Exceptions (if any)
See screenshots and logs above.

### Further technical details
- ASP.NET Core version: .net 50
- Include the output of `dotnet --info`
```
.NET SDK (reflecting any global.json):
 Version:   5.0.100
 Commit:    5044b93829

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19042
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100\

Host (useful for support):
  Version: 5.0.0
  Commit:  cf258a14b7

.NET SDKs installed:
  1.1.13 [C:\Program Files\dotnet\sdk]
  2.1.502 [C:\Program Files\dotnet\sdk]
  2.1.503 [C:\Program Files\dotnet\sdk]
  2.1.504 [C:\Program Files\dotnet\sdk]
  2.1.505 [C:\Program Files\dotnet\sdk]
  2.1.507 [C:\Program Files\dotnet\sdk]
  2.1.508 [C:\Program Files\dotnet\sdk]
  2.1.509 [C:\Program Files\dotnet\sdk]
  2.1.511 [C:\Program Files\dotnet\sdk]
  2.1.512 [C:\Program Files\dotnet\sdk]
  2.1.513 [C:\Program Files\dotnet\sdk]
  2.1.514 [C:\Program Files\dotnet\sdk]
  2.1.515 [C:\Program Files\dotnet\sdk]
  2.1.516 [C:\Program Files\dotnet\sdk]
  2.1.519 [C:\Program Files\dotnet\sdk]
  2.1.600 [C:\Program Files\dotnet\sdk]
  2.1.602 [C:\Program Files\dotnet\sdk]
  2.1.800 [C:\Program Files\dotnet\sdk]
  2.2.101 [C:\Program Files\dotnet\sdk]
  3.0.100 [C:\Program Files\dotnet\sdk]
  3.1.102 [C:\Program Files\dotnet\sdk]
  3.1.201 [C:\Program Files\dotnet\sdk]
  3.1.402 [C:\Program Files\dotnet\sdk]
  5.0.100-rc.1.20452.10 [C:\Program Files\dotnet\sdk]
  5.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.18 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.18 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-rc.1.20451.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.18 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.19 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-rc.1.20451.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.9 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0-rc.1.20452.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```
- The IDE (VS / VS Code/ VS4Mac) as per above

**Comment by dazinator** [comment]
I should also mention that, I have seen a few references to this issue, and apparently [it was fixed in RC2](https://github.com/dotnet/aspnetcore/issues/25887) - but for me, this does not appear to be the case :-/

**Comment by dazinator** [comment]
Additional note: I found what looks to be an identical issue (all be it with not so much detail) reported on the visual studio developer community site, so I also added some detail to it there (like the above logs) https://developercommunity.visualstudio.com/content/problem/1262285/exception-of-type-microsoftvisualstudiosharedvscod.html

**Comment by dazinator** [comment]
Cause: I found setting my default browser (on windows 10, under default apps) to "Chrome" instead of "Edge", and then restarting VS, fixes this issue.

It seems Chrome is a pre-requisite of debugging. I think if this is the case, more should be done to pre-empt this and prompt the user to change their default browser or something similar.

**Comment by mkArtakMSFT** [comment]
@dazinator glad that you've been able to solve this.
The `Edge` browser you have on your machine, is that the new `Chromium`-based edge or the old one?

**Comment by dazinator** [comment]
@mkArtakMSFT 
New chromium based one

**Comment by zmayor** [comment]
Got the same exception with chrome

**Comment by mtorromacco** [comment]
Same exception with Edge (Chromium based)

**Comment by javiercn** [comment]
@dazinator thanks for the additional details.

Does this happen in a brand new project too?

**Comment by damianog** [comment]
Confirm the isuse with Edge Chromium 87.0.664.47, Chrome  86.0.4240.198, but Firefox 82.03 runs.

**Comment by dazinator** [comment]
@javiercn 
I havent tested yet with any brand new projects but its impacting multiple seperare existing projects/ repos.
Also my solution of switching default browser to chrome was a red herring - I now get the same exception occurring with chrome. It seemed to work for a while but then get into a state where this error occurs and just keeps occurring. Perhaps some process is not being managed correctly, not sure.

**Comment by javiercn** [comment]
@dazinator You might perhaps missed some migration steps, that's why it's important to check if an out of the box template works.

**Comment by dazinator** [comment]
@javiercn it does happen for new projects also.
The issue seems to be this:
If the previous debug session doesn't terminate the browser window, and you go to launch a new debug session - I see this error.
If I close the previous window, then launch it works.
Why doesn't the previous window close correctly when you terminate the previous debug session? In my case, if the blazor app has some form of startup error etc.

The screen shot shows the browser window hanging round from first debug session which launched correctly but didn't terminate correctly, and in second tab is the second attempt as launching a debug session, a new window opened and I see the error reported and the debug session terminates, leaving the new orphaned tab. If I close all tabs and start from scratch it works.

![image](https://user-images.githubusercontent.com/3176632/100440601-111c5400-309d-11eb-81ef-4b45517d9b07.png)

**Comment by Flearz** [comment]
I have the same problem, Chrome window opening but no website.
I am wondering if it might be linked to the fact I am using different SSL certificate from the dev one. (Set with https://improveandrepeat.com/2020/05/how-to-change-the-https-certificate-in-iis-express/ )
Or maybe linked to the fact that I use a specific address (changes made in "hosts" file to match 127.0.0.1)
Hope it will be solved soon :)

**Comment by chrisret** [comment]
Had the same problem with Chrome/Edge and netcoreapp3.1 (not 5!). Changing to Firefox worked. It appeared suddenly during work, so I thought my changes in code or the new Resharper feature DPA, which I turned on shortly before, is responsible, but it turned out that deleting the .suo file solved the problem. HTH.

**Comment by mreinhart2805** [comment]
> Had the same problem with Chrome/Edge and netcoreapp3.1 (not 5!). Changing to Firefox worked. It appeared suddenly during work, so I thought my changes in code or the new Resharper feature DPA, which I turned on shortly before, is responsible, but it turned out that deleting the .suo file solved the problem. HTH.

@chrisret Delete of .suo file helped. I have also updated Resharper.

**Comment by hjrb** [comment]
Same here. VS Enterprise 2019 (16.8.3). WASM Blazor project.

**Comment by RahmanM** [comment]
This is totally terrible guys. Was excited to start with Blazor but a total turn off. Just the OOB brand new Blazor Web assembly project with .Net 5 and everything is falling. Getting the same error has the guys above but then changing the Browser to Firefox and at least the page is showing. Clicking the the "Fetch Data" once again failing with "Unhandled Exception" ... "

crit: Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: The provided ContentType is not supported; the supported types are 'application/json' and the structured syntax suffix 'application/+json'.
System.NotSupportedException: The provided ContentType is not supported; the supported types are 'application/json' and the structured syntax suffix 'application/+json'.
  at System.Net.Http.Json.HttpContentJsonExtensions.ValidateContent (System.Net.Http.HttpContent content) <0x2d022c8 + 0x0009e> in <filename unknown>:0 
  at System.Net.Http.Json.HttpContentJsonExtensions.ReadFromJsonAsync[T] (System.Net.Http.HttpContent content, System.Text.Json.JsonSerializerOptions options, System.Threading.CancellationToken cancellationToken) <0x2d020f8 + 0x00006> in <filename unknown>:0 
  at System.Net.Http.Json.HttpClientJsonExtensions.GetFromJsonAsyncCore[T] (System.Threading.Tasks.Task`1[TResult] taskResponse, System.Text.Json.JsonSerializerOptions options, System.Threading.CancellationToken cancellationToken) <0x2c6c1c0 + 0x00150> in <filename unknown>:0 
  at BlazorTodos.Client.Pages.FetchData.OnInitializedAsync () <0x2c16b68 + 0x000f8> in <filename unknown>:0 
  at Microsoft.AspNetCore.Components.ComponentBase.RunInitAndSetParametersAsync () <0x2b0ba90 + 0x00154> in <filename unknown>:0 
  at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask (System.Threading.Tasks.Task taskToHandle) <0x2c6d2e0 + 0x000ca> in <filename unknown>:0 "

**Comment by ramious76** [comment]
I get this same issue.  I get it for both Edge and Chrome but I can run Firefox just fine.  VS(2019) 16.8.3 and Dotnet 5.0.1 Blazor WASM project.

**Comment by aylmercarson** [comment]
same issue with the ootb template, with authentication enabled. seems now to work in firefox, nothing else. yet it worked yesterday in edge. bit of a magical mystery tour :(

**Comment by MeesvanStraten** [comment]
Problem still here, mainly when launching as server instead of using ISS profile.
New projects work but when switching to a different launch profile it happens.
Error does not get thrown immediately but after 15-20 seconds.

**Comment by vnbaaij** [comment]
What fixes this for me is:
-  Close Visual Studio
-  Delete the "debuggerConfiguration-{guid}.json" file from the "{solution}\.vs\{solutionname}\config\" folder .
- Start Visual Studio and happy debugging!

**Comment by scottkuhl** [comment]
This happens to me when I disable Anonymous Authentication and enable Windows Authentication on a new Blazor Web Assembly project hosted in ASP.NET.  I can set launch browser to false and put the url in a browser window manually and it works.  Reverting the server project back to Anonymous Authentication also fixes the issue.

**Comment by adamgusky** [comment]
I am having this issue as well. @vnbaaij - which browser did deleting the debuggerConfiguration file work with?

**Comment by vnbaaij** [comment]
@adamgusky for me this worked on Edge (Development version)

**Comment by MeesvanStraten** [comment]
I tested it more today with different browsers, Chrome, Edge(chromium) all throw the error after about 15 seconds.
Mozilla Firefox Developer edition does work, runs for more than 10 minutes succesfully now.

**Comment by Flearz** [comment]
Still the same problem with .NET 5
Maybe is it due to application of a specific certificate by changing IIS Express options 
https://improveandrepeat.com/2020/05/how-to-change-the-https-certificate-in-iis-express/
https://improveandrepeat.com/2017/07/how-to-configure-iis-express-to-accept-ssl-client-certificates/
I don't know what to do, it is not happening with a new blazor project template, only with my current project.
Firefox is launching without problem but the certificate is not trusted and my application not working properly.

**Comment by aventus13** [comment]
I just ran into the same issue- Seemingly randomly as I haven't made any major updates since yesterday. Doing something similar to what @vnbaaij suggested fixed the problem. In my case I deleted the entire .vs folder, cleaned the solution, rebuilt and launched it again.

**Comment by brunolpw** [comment]
Any solution?

I have the same problem.  😞

**Comment by brunolpw** [comment]
> I just ran into the same issue- Seemingly randomly as I haven't made any major updates since yesterday. Doing something similar to what @vnbaaij suggested fixed the problem. In my case I deleted the entire .vs folder, cleaned the solution, rebuilt and launched it again.

I did the same and it's work now!
Before, I tried onli cleaned the solution and rebuild, but not delete the ".vs" folder.

**Comment by peterthomastlc** [comment]
I'm getting the same thing, in both Chrome and Edge. Cleaning and deleting the .vs folder makes no difference.
The order of catastrophe when I try to debug is:-
1) The solution builds and a new Chrome window opens, with an 'about:blank' tab and and a blank window.
    
2) Nothing happens for another 60 seconds. Then a message box is shown with:
    "Failed to launch debug adapter...Excpetion of type 'Microsoft.VisualSrudio.Shared.VSCodeDebugProtocol.ProtocolException' was thrown."
    Checking visualstudio-js-debugger.txt I see:
    `{"tag":"runtime.launch","timestamp":1610189750506,"message":"Launch returned error","metadata":{"error":{"message":"Unable to launch browser: \"Could not open 
ws://localhost:53414/_framework/debug/ws-proxy?browser=ws%3A%2F%2Flocalhost%3A57745%2Fdevtools%2Fbrowser%2F4933e2b3-f339-4f8f-9b4f-c92f28a43a8f\"","stack":"Error: Unable to launch browser: \"Could not open ws://localhost:53414/_framework/debug/ws-proxy?browser=ws%3A%2F%2Flocalhost%3A57745%2Fdevtools%2Fbrowser%2F4933e2b3-f339-4f8f-9b4f-c92f28a43a8f\"\n    at _.prepareLaunch (c:\\program files (x86)\\microsoft visual studio\\2019\\professional\\common7\\ide\\commonextensions\\microsoft\\jsdiagnostics\\debugger\\debugAdapter\\out\\src\\targets\\browser\\browserLauncher.js:102:19)"},"wasCancelled":true,"name":"_"},"level":2}
`
followed by
`{"tag":"dap.send","timestamp":1610189753318,"metadata":{"connectionId":0,"message":{"seq":10,"type":"event","event":"output","body":{"category":"telemetry","output":"js-debug/error","data":{"!error":{"error":{"message":"read ECONNRESET","name":"Error","stack":"Error: read ECONNRESET\n    at TCP.onStreamRead (internal/stream_base_commons.js:205:27)"}},"error":{"error":{"message":"read ECONNRESET","name":"Error","stack":"Error: read ECONNRESET\n    at TCP.onStreamRead (internal/stream_base_commons.js:205:27)"}},"exceptionType":"uncaughtException"}}}},"level":0}
`
My environment is:
VS2019 16.8.3
ASP.Net Core 5.0
Docker Desktop, with wsl2 components installed (changing between wsl2 & wsl makes no difference)

This is happening repeatedly for me, and is making Blazor unusable as a development platform - which is a huge shame as it promises so much.
Incidentally, I also commonly get an error when debugging Blazor apps in IIS Express, where it's unable to start IIS Express. Again, it seems OK on newly created projects, but a few debugging sessions later it rears it's ugly head and is then impossible to get rid of - cleaning the solution, deleting the .vs folder & restarting VS doesn't help. Not sure if this error is related to the above.

**Comment by chrisret** [comment]
@peterthomastlc: "unable to start IIS Express" usually you see, when there is a port your project uses is blocked by another process. Just try changing the port(s) in your launchsettings: "iisSettings":"iisExpress":"applicationUrl", sometimes you need to change sslPort, too. To me it is necessary to choose a port number at least 100 or even 1000 or more different from the existing, since on my machine the problem occurs due to a stupid virus protection tool, which is required to meet company standards (sigh).

**Comment by chrisret** [comment]
Back from vacation now, and the problem has vanished.
I have two projects that evolved this problem. One of them I managed to repair by deleting the .suo, as described on Dec. 16. The other couldn't be cured by this method. But after not touching it for 3 weeks, the problem disappeared by itself. Working stable in Chrome, Edge an Firefox since yesterday.
Now I wonder, if some certificate expired and was replaced silently?

**Comment by ssprencel** [comment]
> What fixes this for me is:
> 
> * Close Visual Studio
> * Delete the "debuggerConfiguration-{guid}.json" file from the "{solution}\.vs\{solutionname}\config\" folder .
> * Start Visual Studio and happy debugging!

Thank you @vnbaaij!  That fixed it for me too!

**Comment by jeremyknight-me** [comment]
> What fixes this for me is:
> 
> * Close Visual Studio
> * Delete the "debuggerConfiguration-{guid}.json" file from the "{solution}\.vs\{solutionname}\config\" folder .
> * Start Visual Studio and happy debugging!

Wasn't working in Chrome or Edge (Chromium) and this fixed it! Thanks!!!

**Comment by Flearz** [comment]
> 
> 
> > What fixes this for me is:
> > 
> > * Close Visual Studio
> > * Delete the "debuggerConfiguration-{guid}.json" file from the "{solution}.vs{solutionname}\config" folder .
> > * Start Visual Studio and happy debugging!
> 
> Wasn't working in Chrome or Edge (Chromium) and this fixed it! Thanks @vnbaaij!!!!

Not for me unfortunately

**Comment by aventus13** [comment]
@Flearz did you try deleting the entire .vs folder as I suggested above?

**Comment by martinmueller4voice** [comment]
Deleting the debuggerConfiguration-{guid}.json fixed the problem for me as well!!!
Thank you so much!

**Comment by azzuro123** [comment]
Debugging works fine, I had the same issue after migrating my Blazor Wasm Hosted app to .NET5.0. I did the following:

- Delete the .vs folder in the project as described above
- Make sure that the .UseWebAssemblyDebugging(); is active in your environment

**Comment by JoeMPoon** [comment]
> What fixes this for me is:
> 
>     * Close Visual Studio
>     * Delete the "debuggerConfiguration-{guid}.json" file from the "{solution}\.vs\{solutionname}\config\" folder .
>     * Start Visual Studio and happy debugging!

THANKS!! my Blazor WebAssemlby debugging works again! Thanks a lot!!

**Comment by elylv** [comment]
I upgraded to Win10 1909, and since I was in an upgrading mood, I upgraded to VS2019 16.4 also.
When I first tried to debug my existing app, IIS Express was complaining that the port was in use.  Seems Windows 1909 had reserved some ports in the range I was using.  I changed the port to something not in use/reserved, and now I'm getting this error with the debug adapter.

I've tried deleting the .vs folder completely, cleaning solution, and debugging, but no success.  Any more suggestions?  I literally can't get on with my work because of this.

**Comment by elylv** [comment]
Just to add, I have the VS2019 v16.8.0 Preview installed on my machine from a while back, and the issue persists with this version of VS also, so it would seem to point to the Win10 1909 update being the culprit.

**Comment by martinmueller4voice** [comment]
I don’t think Win 10 b1909 is to blame since I’m running 20H2 and had the problem with 2004 as well.
Deleting the debuggerConfiguration*.json file worked, though.

Von: Stephen McMahon <notifications@github.com>
Gesendet: Montag, 25. Januar 2021 12:46
An: dotnet/aspnetcore <aspnetcore@noreply.github.com>
Cc: Martin Müller <Martin.Mueller@4voice.de>; Comment <comment@noreply.github.com>
Betreff: Re: [dotnet/aspnetcore] Microsoft Visual Studio Failed to launch debug adapter 'Microsoft.VisualStudio.Shared.VSCodeDebugProtocol.ProtocolException (#28113)


Just to add, I have the VS2019 v16.8.0 Preview installed on my machine from a while back, and the issue persists with this version of VS also, so it would seem to point to the Win10 1909 update being the culprit.

**Comment by elylv** [comment]
Yeah, as I mentioned, that fix is not working for me at all, and I'm now stuck, unable to do my job.  Issue only started occurring after 1909 was pushed out to my system.

**Comment by martinmueller4voice** [comment]
Does debugging a new project work?
If so, there _must_ be some remains in your existing project keeping the debugger from starting correctly.

Btw. using a terrorist organization's logo as your profile picture is inappropriate. I suggest you change it, otherwise github might take actions.

**Comment by elylv** [comment]
Brand new Blazor solutions, with absolutely no changes, also have this problem.

**Comment by martinmueller4voice** [comment]
That's why deleting the .vs folder didn't change a thing.
You could try a different default browser, but apart from that I'm out of ideas.

**Comment by elylv** [comment]
Running it in .Server mode launches the browser and loads the page properly, but it's not possible to debug any client-side code doing it that way.  I suppose it will have to do for now.

Would be good to hear from MS that they actually understand what the issue is, and we don't get to the "next sprint planning" and they fix something else instead, and we report the issue again, and it will be looked at in the next next sprint planning!

**Comment by mkArtakMSFT** [comment]
This is a VS specific issue, and we will be following-up internally with the appropriate teams to  get them to handle this. Thanks everyone for all the information.

**Comment by lambiuk** [comment]
Hi all, after fighting with that one for a couple of days, in our case we were getting this error on some of our dev environments but not all and it turned out to be a **missing IIS Express Certificate**.

Debugging would work when running without SSL.

We found out by going to "MMC > Current User Certs > Trusted Root Certification Authorities" and there should be a valid **IIS Express Development Certificate** there. If you cannot find one you can follow the instructions here to reset it:
https://improveandrepeat.com/2020/05/recreate-the-self-signed-https-certificate-for-localhost-in-iis-express/

Hope it helps someone!

**Comment by elylv** [comment]
Not that it's really a solution, but I got a new system today, cleanly imaged with Win1909.  Installed VS16.8.4, cloned my repo, and it's debugging without issue.  Perhaps a clean reinstall of VS would fix it, or surely a clean reinstall of Windows.  I wouldn't be doing that, though, and I wouldn't blame anyone else for not doing it either!

**Comment by djary** [comment]
same on VS 2019 16.8.4, blazor webassembly

**Comment by martinmueller4voice** [comment]
Happened again today.
Yesterday I could start debugging the Blazor project with Edge as browser without problem, today I get the dreaded ProtocolException again. (VS 2019 16.8.4)
Removing the .vs\<MyProjectName>\config folder and re-loading VS again fixed the problem for me, though.

**Comment by sks-ttaylor** [comment]
Happened to us:  Blazor WASM;  issue was following the MS instructions on adding custom web.config for deployment. renaming web.config worked for us.  <PublishIISAssets>true</PublishIISAssets>

**Comment by joer33304** [comment]
I tried all the work around, I cannot get rid of the startup error. Interesting observation maybe, the browser url does not get filled with app url set in the startup settings. So something fails before it even gets there

**Comment by joer33304** [comment]
> Hi all, after fighting with that one for a couple of days, in our case we were getting this error on some of our dev environments but not all and it turned out to be a **missing IIS Express Certificate**.
> 
> Debugging would work when running without SSL.
> 
> We found out by going to "MMC > Current User Certs > Trusted Root Certification Authorities" and there should be a valid **IIS Express Development Certificate** there. If you cannot find one you can follow the instructions here to reset it:
> https://improveandrepeat.com/2020/05/recreate-the-self-signed-https-certificate-for-localhost-in-iis-express/
> 
> Hope it helps someone.
> 
> PS: Big thanks to @SteveSandersonMS for a very promising framework, the entire dev team is loving Blazor here!

I switched off SSL completely for the app, still get the same error

**Comment by ryandeering** [comment]
I've been suffering with this issue for a while after upgrading to .NET 5. It can come and go, but it's still super annoying. Tried deleting .VC, turning off SSL... the only temporary fix is clearing the browser cache of Chrome and Edge and switching between the two. It is worth noting everything is a lot more stable on Firefox due to the lack of WASM debugging support. It is also worth noting it seems to take up a lot less memory too, starting my solution (ASP.NET, IdentityServer, Blazor WASM) with Firefox as well.

**Comment by Denmarkstan** [comment]
I tried many suggestion but none helped except @vnbaaij solution. Thanks @vnbaaij!!!!

**Comment by rad765** [comment]
I see the "about:blanc" problem with brand new solutions in VS 16.9.0 Preview and .Net v6.0.0-preview1. 
None of the suggestions over solves for me.

Downgrading Target Framework to .Net 5.0 and the NuGet Packets to 5.0.3 sort of "solved it". 
Upgrading again to .Net 6.0 - "about:blanc" problem is back. 

After a while I also get the dialog with "Failed to launch debug adapter"

**Comment by MichaelBakkerDP** [comment]
I encountered this problem after installing Visual Studio 2019 Community Preview (16.9.0 Preview 3 and later Preview 4, which didn't solve it) next to my VS 2019 Professional Install (16.8.5) already present.
No information present in Output window, ...\Local\Temp\VSTelem.Out or event logs.

Deleting the .vs folder from the problematic solution worked for me.

**Comment by bkwdesign** [comment]
I also am getting this on my .NET Framework web app 
3 month old Wind 10 machine doesn't have Chrome. Only chromium-Edge installed.

I have two startup web projects (web api and an MVC app). Both launch in separate browser windows and were working totally fine.

I tried to put a breakpoint into the an open javascipt file in the IDE which prompted me with a 3-choice dialog to the effect of "enable javascript debugging and restart" or "continue debugging" ..etc - I clicked the first option and that's when my troubles started.

**Comment by leigh-pointer** [comment]
My solution was simple and after trying so many different suggestions I just deleted the complete .vs<MyProjectName>\ restarted the solution and was back up and debugging!

**Comment by roy-cohen-2005** [comment]
I had the same bug with chrome, I switched to Edge to check if that'll help but it didn't. When I switched back to chrome it worked. I don't why but switching the browser to Edge and back fixed the issue.
To be clear, I didn't change the default browser, instead I clicked the gray triangle next to the text "IIS Express", then "Web Browser" and finally chose a browser.

**Comment by saddlebear** [comment]
I get this error when ASPNETCORE_ENVIRONMENT is set to "Production" but the Solution Configuration drop-down (left of "Any CPU" dropdown) is set to Debug and not Release.   Note, if set to Release I also have to use Debug > Start Without Debugging (Ctrl+F5) or I am put into some sort of dialog hell.   IIS Express works fine then but Kestrel will not start (HTTP 500 error) because Identity Server wants a key for token transmission in production environments.   Hope this helps.

**Comment by ldsenow** [comment]
I have the same issue as well. Tried all of above unfortunately none of them work for me. I am on 16.9 .net 5 blazor webassembly.
It is a bit annoying.

**Comment by niko-la-petrovic** [comment]
Deleting the config folder from .vs and restarting VS helped me multiple times.

**Comment by Flearz** [comment]
Still having the problem, I thinks it's really related to custom SSL settings set with https://improveandrepeat.com/2020/05/how-to-change-the-https-certificate-in-iis-express/ .

**Comment by sdudnic** [comment]
please see the resmonitor (resmon.exe) Visual Studio for a unknown reason will launch the browser **twice** for the configured port.
![image](https://user-images.githubusercontent.com/35489219/109876298-e8929000-7c71-11eb-9c8a-a0336bd43cc2.png)

see and this
https://stackoverflow.com/questions/66456892/failed-to-launch-debug-adapter-vs-tries-to-connect-to-a-random-port


Why does it start twice the process (on the same port)?

**Comment by niko-la-petrovic** [comment]
> Still having the problem, I thinks it's really related to custom SSL settings set with https://improveandrepeat.com/2020/05/how-to-change-the-https-certificate-in-iis-express/ .

I've experienced the issue even when not running the project with IIS Express. Could still be related to SSL certs.

**Comment by andrewsmalek** [comment]
I've tried all the above solutions and nothing worked for me.

However, **Disabling "Script Debugging" seems to fix the issue** for me. 

![image](https://user-images.githubusercontent.com/11490232/110011421-189c6a80-7d28-11eb-91e2-bd9d9eed88b0.png)

**Comment by ldsenow** [comment]
At the end,  I found out if I add the pathbase environment variable,  the debugger wont be attached for some reason.

![image](https://user-images.githubusercontent.com/177045/110059593-a162e700-7db8-11eb-8f97-cd48b11da5e6.png)

**Comment by HanyoVladScarlet** [comment]
I guess there's something wrong with firewall. Cuz I feels like that I did something wrong while checking the authority of the project with network settings.

I catch this problem every time I launch the project with VPN on, while everything goes smoothly if I turn VPN off.

**Comment by RNONGFU001** [comment]
yes, disable script debugging, above mentioned by andrewsmalek works for me. That is great, finally get rid of this annoying bug, before that, had to end the processes of edge first in Task Manager.....

**Comment by rakatyal** [comment]
Hello everyone!

Huge apologies to all of you for the bad experiences. The error dialog box posted by the author of this issue is a generic one and can pop up due to a number of underlying issues. Due to this, it appears there are multiple bugs/issues in play on this thread.

We have identified some of these issues and are in the process of fixing them in the next VS Preview (16.10 Preview 2). I would like to mention some of the prominent ones here for some clarity:

1. We have identified an issue with the latest version of Edge which is causing the VS JavaScript debugger to display that error. To fix this, [please follow the workaround mentioned here](https://developercommunity.visualstudio.com/t/Debugger-wont-start-second-time-with-VS/1355978?viewtype=all).

2. There is another issue with the blazor and some other dotnet apps created through the command line and trying to launch the debugger. This only happens with an "unsaved solution". The command line apps sometimes do not have a .sln file (VS solution file) created with it and this again causes this error to pop up. To fix this, please save your solution from inside VS to create the .sln file and try again.

3. If you are not interested in debugging JavaScript, you can disable 'Script Debugging' as mentioned in one of the above comments. This should also get rid of this error.

Once 16.10 Preview 2 is out, please enable the new debugger by selecting this Tools->Option. This has some other bug fixes too.

![image](https://user-images.githubusercontent.com/12533467/110406102-869da600-8036-11eb-91a8-b52957678119.png)

Lastly, if there are still other debugging issues even after trying out the next preview, please [report the problem from inside VS](https://developercommunity.visualstudio.com/report?viewtype=all&space=8&entry=problem). This provides us the right logs and diagnostics to address them.

**Comment by RonPeters** [comment]
None of these suggestions worked for me. But this did: https://developercommunity.visualstudio.com/t/exception-of-type-microsoftvisualstudiosharedvscod/1262285#T-N1315474

"I face the same issue here. If I close all browsers instance and delete file /.vs/[project name]/config/debuggerConfiguration-{guid}.json it works."

Deleting that debuggerConfiguration file was the solution for me

**Comment by mgwilliam** [comment]
I seem to get this each time I update VS (most recently to 16.9.1)

Deleting the debuggerConfiguration file works for me

**Comment by Hectop2021** [comment]
Using Microsoft Visual Studio Community 2019 Version 16.9.2 and Windows Edge Version 89.0.774.57 (Official build) (64-bit)

Error experienced when trying to start a second application while one is already running.
Applications have different ports configured and are not the same code.

Tried deleting the debuggerConfiguration file in both applications prior to starting.  After they start, both file names are the same.
debuggerConfiguration-2039bca5-38d4-4a63-ae2b-b09bd3ca3554.json
Any idea where the guid portion comes from?

Have also enabled debugging as described above.  Will try to send feedback.

**Comment by cuongvu0203** [comment]
> I've tried all the above solutions and nothing worked for me.
> 
> However, **Disabling "Script Debugging" seems to fix the issue** for me.
> **Update:** as mentioned by RNONGFU001, make sure to end the active task for your browser first (chrome, edge...etc.).
> 
> ![image](https://user-images.githubusercontent.com/11490232/110011421-189c6a80-7d28-11eb-91e2-bd9d9eed88b0.png)

try this solution! It work with me!

**Comment by tombauto** [comment]
> I encountered this problem after installing Visual Studio 2019 Community Preview (16.9.0 Preview 3 and later Preview 4, which didn't solve it) next to my VS 2019 Professional Install (16.8.5) already present.
> No information present in Output window, ...\Local\Temp\VSTelem.Out or event logs.
> 
> Deleting the .vs folder from the problematic solution worked for me.

Thanks for this suggestion! Worked for me. I was very surprised that I just had that issue wherein just last night everything is working fine.

**Comment by ylr-research** [comment]
Same issue here. 
- VS 2019 Community Edition. 
- Windows 10. .NET 6 Preview. 
- Brand new project targeting .NET 6. 
I tried removing `vs` folder and switching to Chrome and still get the same error.

**Comment by pupper9** [comment]
Same issue, after starting the project it sits at about:blank and after 30 seconds I get the "Failed to launch debug adapter" error.

My projects where working fine 3 hours ago and suddenly stopped working.

I tried the following and **nothing** works:

- Delete  /.vs/[project name]/config/debuggerConfiguration-{guid}.json
- Delete .vs folder
- Disable SSL
- Start a brand new project with the default template
- VS2019 16.10 Preview 1
- Completely remove VS2019 and make a fresh install without importing previous settings
- Chrome as default browser
- Edge as default browser

Here's the only thing that works:

- Use Firefox withoud debugging.

**Comment by MarcoTinner** [comment]
Same issue after the todays update of VS using Web-Browser Google Chrome or Microsoft Edge.
Solution for me works if change Web-Browser to any other like FireFox or SlimBrowser .or. 
remove Launch browser in Project properties / Debug.
![image](https://user-images.githubusercontent.com/16720951/114360134-e5f35680-9b9e-11eb-9f99-e4cdf09cefcd.png)

**Comment by askew** [comment]
I've just been hit my this issue after picking up a project (Blazor WASM) that I haven't worked on for a few months. Deleting the `.vs` folder, as mentioned by @MichaelBakkerDP seems to have fixed it for me.

**Comment by AhmedHemdan21** [comment]
> I've just been hit my this issue after picking up a project (Blazor WASM) that I haven't worked on for a few months. Deleting the `.vs` folder, as mentioned by @MichaelBakkerDP seems to have fixed it for me.

Thanks to @askew comment, deleting the .vs folder and use Chrome instead of Edge has foxed the issue for me.

**Comment by KurtGarrettALS** [comment]
Here is what worked for me.
Add UseWebAssemblyDebugging() to Startup. See this article "Debug ASP.NET Core Blazor WebAssembly" on Microsoft site.

![image](https://user-images.githubusercontent.com/31772255/115095165-34318e00-9eee-11eb-8982-cc8ca4c11487.png)

**Comment by ylr-research** [comment]
Long story short. None of the things suggested so far worked for me. One thing I noticed though, if you have many projects, you may think having your WASM project set as the Startup is enough. Well, let me tell you, I bet your debugger is set to **IIS Express**. In the debugger dropdown change from **IIS Express** to `{Your WASM project name}` and voila! you got it running.

![image](https://user-images.githubusercontent.com/63487874/115097355-904ce000-9ef7-11eb-8354-81d5f718c211.png)

**Comment by erikthysell** [comment]
Neither IIS express nor the other one works...

**Comment by alexwhite68** [comment]
Tried all of the above, nothing works, tried a brand new WASM template, it does not work out of the box, it will not stop on any breakpoints I am running:- 

Microsoft Visual Studio Community 2019 Version 16.9.4

I have Edge, Chrome and Firefox installed none of them stop on the breakpoints, all fire up the app properly but no breakpoints are hit.

I have tried cleaning the projects, deleting the bin and obj folders from all projects, deleting the .vs folder, I have script debugging off, all projects are set to debug, the app works but debugging is not hitting any breakpoints at all.

**Comment by AlexW68** [comment]
I have now installed a completely fresh version of windows 10, updated it, installed visual studio community edition, created an app from the WASM template and it does not hit the breakpoints, this is now 3 PC's and one MAC not debugging ASP.NET projects at all. 

Steps to reproduce install all of the above create a new app based on the WASM template, stick a break point on the following line.

forecasts = await Http.GetFromJsonAsync<WeatherForecast[]>("WeatherForecast");

it loads that data without breaking on the breakpoint.

**Comment by AlexW68** [comment]
Hi Steve, 

Thanks for the reply, it hits the breakpoints on a new project new vm, old project on an old computer if I copy the counter.razor component into the project and set a breakpoint.

So that works sometimes I get the following error bubble up through to the browser 

Unhandled exception rendering component: '<' is an invalid start of a value. Path: $ | LineNumber: 0 | BytePositionInLine: 0. System.Text.Json.JsonException: '<' is an invalid start of a value. Path: $ | LineNumber: 0 | BytePositionInLine: 0. ---> System.Text.Json.JsonReaderException: '<' is an invalid start of a value. LineNumber: 0 | BytePositionInLine: 0

I can set a breakpoint and hit it in the API function server side that seems to work ok. What I am also not able to do when this seems to happen is try catch blocks round the API calls and capture the error it does not seem to want to bubble the error into the client side WASM for some reason.

I did note (and this maybe something completely different) that in another project I set breakpoints, they hit every now and again, then stop debugging and restart and the breakpoints are then not solid red but a red outlined dot, once you are into that module/component you can click it twice to make it solid red.

Thanks for your help I know you guys are busy, but this is really busted at the moment.

(don't think it matters but all the firewalls are switched off)

**Comment by AlexW68** [comment]
Testing this some more that project has now failed, the counter breakpoint still works, but I am finding that if the breakpoint is set inside the api (server app) the breakpoint may get hit on the client side.

In the past clearing the bin/obj folders has helped with these types of issues, this does not seem to work now.

Also changing the api code to the following seems to make the breakpoints more reliable

```
var rng = new Random();
WeatherForecast[] wf = Enumerable.Range(1, 5).Select(index => new WeatherForecast
{
Date = DateTime.Now.AddDays(index),
TemperatureC = rng.Next(-20, 55),
Summary = Summaries[rng.Next(Summaries.Length)]
})
.ToArray();
return wf;
```

This way I can see what is being returned before it's passed back to the client.

**Comment by tadeutorquato** [comment]
Hi guys,

I tried all the workarounds and the only one that worked for me successfully was to remove .vs from the project folder.

**Comment by giarico** [comment]
> Same issue for me.
> 
> Visual Studio Entrerprise 16.9.4
> NET 5
> Edge (Chromium) & Chrome
> 
> With IISExpress the browser is stuck on the "about:blank" page and after a while I get the following error:
> "Failed to launch debug adapter".
> 
> I have tried all workarounds without success.

Just to give some more detail, this is the error i get in "visualstudio-js-debugger.txt" file:
Error: Unable to launch browser: \"Could not open ws://localhost:22410/_framework/debug/ws-proxy?browser=ws%3A%2F%2Flocalhost%3A59874%2Fdevtools%2Fbrowser%2F17a7e068-58d4-415a-a333-e6de0adb6e99\"

**Comment by AlexW68** [comment]
Tried the preview version of Visual Studio problem still exists there too.

Here is a small video, this is a basic windows 10 VM with visual studio preview on it, the https and other options really make no difference to if it's going to work reliably or not, I am happy working with alpha builds and helping working through this problem, if I am guided to test certain things.

**Comment by emamcoder** [comment]
> @dazinator thanks for the additional details.
> 
> Does this happen in a brand new project too?

Yes, this is with a brand new project too.

**Comment by eannelson** [comment]
I get the same thing -- deleting the .vs folder within the solution and restarting Visual Studio 2019 seems to fix it ...eventually. Based on the logs, when this happens the debugger is usually trying to attach to a port that I never specified (even in other recent solutions). 

System.TimeoutException: Failed to start the debug proxy within the timeout period of 10 seconds.
   at Microsoft.AspNetCore.Builder.DebugProxyLauncher.LaunchAndGetUrl(IServiceProvider serviceProvider)
   at Microsoft.AspNetCore.Builder.WebAssemblyNetDebugProxyAppBuilderExtensions.<>c.<<UseWebAssemblyDebugging>b__0_1>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Builder.Extensions.MapMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request finished in 6.5718ms 500 text/plain
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request starting HTTP/1.1 GET https://localhost/_framework/debug/ws-proxy?browser=ws%3A%2F%2Flocalhost%3A64613%2Fdevtools%2Fbrowser%2F9e594754-e6fe-4266-bf05-cdd31c776220  
Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware: Error: An unhandled exception has occurred while executing the request.

**Comment by JoelW187** [comment]
Just upgraded to VS 16.10.1 and I'm still getting this error frequently when launching a debug session. It happens on both Core (Blazor WebAssembly) projects and older Asp.Net Framework (4.x) projects. I've tried all the usual -- delete .vs folder, delete bin/obj folders, etc. There seems to be no pattern to it in my case.

**Comment by exerror14** [comment]
[[Solved] “Failed to load resource: net::ERR_HTTP2_PROTOCOL_ERROR” for React app after upgrading to Visual Studio 2019 16.10.0](https://exerror.com/failed-to-load-resource-neterr-http2-protocol-error-for-react-app-after-upg/)

**Comment by danroth27** [comment]
Hi folks. If you hit this "Microsoft Visual Studio Failed to launch debug adapter" due to a ProtocolException when trying to debug a Blazor WebAssembly app, please be sure the use the Visual Studio Send Feedback -> Report a Problem feature to let us know. This will automatically collect the logs that we need to understand why the the debug proxy failed to launch. Thanks for your patience on this one!

**Comment by danroth27** [comment]
Since this issue isn't actionable on our side without the corresponding VS logs, I'm going to close this issue and we'll follow up on the various Visual Studio Feedback issues. Hopefully we'll get these issues sorted soon.

**Comment by mohamedalikhalifi** [comment]
> Hi folks. If you hit this "Microsoft Visual Studio Failed to launch debug adapter" due to a ProtocolException when trying to debug a Blazor WebAssembly app, please be sure the use the Visual Studio Send Feedback -> Report a Problem feature to let us know. This will automatically collect the logs that we need to understand why the the debug proxy failed to launch. Thanks for your patience on this one!

This error happened many times to me recently and there is no such a button to send the feedback in the error dialog.

**Comment by danroth27** [comment]
@mohamedalikhalifi The Send Feedback button should be in the upper right-hand corner of VS:

![image](https://user-images.githubusercontent.com/1874516/124634703-7dcdad00-de3b-11eb-9022-530a23b49f29.png)

**Comment by ErnanMaisels** [comment]
I having the same problem

**Comment by ErnanMaisels** [comment]
In my case, it was helpful to clear cached files
![image](https://user-images.githubusercontent.com/76112655/127106442-5ab9cbf8-0dcd-418d-99cc-56cc4cafacef.png)

**Comment by FastTracOnline** [comment]
Submitted using Feedback, All requested files and then some.  I cannot recommend Blazor to my employer at this time.  I will not recommend a product that cannot be debugged.

**Comment by Shanair** [comment]
I read through the whole postings to figure out what is going on here.
It seems the the "only" problem is to automaticly start a browser with the debug button.

Workarround (just to go on working :) ) :
In your **launchSettings.json** set the **launchBrowser** to **false**
Start the project and manually open your url in the browser => localhost:5000

Shame... this issue is up over a year now... Also in current version of VS + new Project
It doesn't appear for every user. My friend tested it also and got no issue...

**Comment by MS-Hayden** [comment]
I've found that unchecking the proxy server checkbox in LAN settings fixes this. I also check automatic detection.

This particular thing has caused this twice for me... completely out the blue...

![image](https://user-images.githubusercontent.com/77356746/131836204-8f31e11e-1963-45a9-bd51-7025604d79e7.png)

**Comment by dazinator** [comment]
> I've found that unchecking the proxy server checkbox in LAN settings fixes this. I also check automatic detection.
> 
> This particular thing has caused this twice for me... completely out the blue...
> 
> ![image](https://user-images.githubusercontent.com/77356746/131836204-8f31e11e-1963-45a9-bd51-7025604d79e7.png)

@MS-Hayden - do you use fiddler or similar tool for inspecting local network traffic? I beleive I found in the past if you don't stop fiddler properly it can leave that tickbox checked.
